beacon-chain/db/kv/state.go

```diff
@@ -686,134 +667,6 @@ func (s *Store) slotByBlockRoot(ctx context.Context, tx *bolt.Tx, blockRoot []by
 	return stateSummary.Slot, nil
 }
 
-// HighestSlotStatesBelow returns the states with the highest slot below the input slot
-// from the db. Ideally there should just be one state per slot, but given validator
-// can double propose, a single slot could have multiple block roots and
-// results states. This returns a list of states.
-func (s *Store) HighestSlotStatesBelow(ctx context.Context, slot types.Slot) ([]state.ReadOnlyBeaconState, error) {
-	ctx, span := trace.StartSpan(ctx, "BeaconDB.HighestSlotStatesBelow")
-	defer span.End()
-
-	var best []byte
-	if err := s.db.View(func(tx *bolt.Tx) error {
-		bkt := tx.Bucket(stateSlotIndicesBucket)
-		c := bkt.Cursor()
-		for s, root := c.First(); s != nil; s, root = c.Next() {
-			if ctx.Err() != nil {
-				return ctx.Err()
-			}
-			key := bytesutil.BytesToSlotBigEndian(s)
-			if root == nil {
-				continue
-			}
-			if key >= slot {
-				break
-			}
-			best = root
-		}
-		return nil
-	}); err != nil {
-		return nil, err
-	}
-
-	var st state.ReadOnlyBeaconState
-	var err error
-	if best != nil {
-		st, err = s.State(ctx, bytesutil.ToBytes32(best))
-		if err != nil {
-			return nil, err
-		}
-	}
-	if st == nil || st.IsNil() {
-		st, err = s.GenesisState(ctx)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	return []state.ReadOnlyBeaconState{st}, nil
-}
-
-// createStateIndicesFromStateSlot takes in a state slot and returns
-// a map of bolt DB index buckets corresponding to each particular key for indices for
-// data, such as (shard indices bucket -> shard 5).
-func createStateIndicesFromStateSlot(ctx context.Context, slot types.Slot) map[string][]byte {
-	ctx, span := trace.StartSpan(ctx, "BeaconDB.createStateIndicesFromState")
-	defer span.End()
-	indicesByBucket := make(map[string][]byte)
-	// Every index has a unique bucket for fast, binary-search
-	// range scans for filtering across keys.
-	buckets := [][]byte{
-		stateSlotIndicesBucket,
-	}
-
-	indices := [][]byte{
-		bytesutil.SlotToBytesBigEndian(slot),
-	}
-	for i := 0; i < len(buckets); i++ {
-		indicesByBucket[string(buckets[i])] = indices[i]
-	}
-	return indicesByBucket
-}
-
-// CleanUpDirtyStates removes states in DB that falls to under archived point interval rules.
-// Only following states would be kept:
-// 1.) state_slot % archived_interval == 0. (e.g. archived_interval=2048, states with slot 2048, 4096... etc)
-// 2.) archived_interval - archived_interval/3 < state_slot % archived_interval
-//   (e.g. archived_interval=2048, states with slots after 1365).
-//   This is to tolerate skip slots. Not every state lays on the boundary.
-// 3.) state with current finalized root
-// 4.) unfinalized States
-func (s *Store) CleanUpDirtyStates(ctx context.Context, slotsPerArchivedPoint types.Slot) error {
```

Comment:
I'm particularly interested in eyes on this removal. I don't know if there are other areas which could be writing states that require cleanup, perhaps removing this exposes some issues? If we're concerned about that possibility, one solution could be to leave this running but `log.Error` when it finds a dirty state.